packages/next/next-server/lib/router/utils/parse-relative-url.ts

```diff
@@ -0,0 +1,24 @@
+const DUMMY_BASE = new URL('http://n')
+
+/**
+ * Parses path-relative urls (e.g. `/hello/world?foo=bar`). If url isn't path-relative
+ * (e.g. `./hello`) then at least base must be.
+ * Absolute urls are rejected.
+ */
+export function parseRelativeUrl(url: string, base?: string) {
+  const resolvedBase = base ? new URL(base, DUMMY_BASE) : DUMMY_BASE
+  const { pathname, searchParams, search, hash, href, origin } = new URL(
+    url,
+    resolvedBase
+  )
+  if (origin !== DUMMY_BASE.origin) {
+    throw new Error('Invalid relative URL')
```

Comment:
When can this case be hit? Does it need to be more user friendly or is this an invariant?

Reply:
When somebody passes an absolute url, like

```js
parseRelativeUrl('http://google.com/hello')
```
It's been handled upstream, with a more user friendly error though: https://github.com/vercel/next.js/pull/14827/files#diff-5eff10c96fa6858c659318492c4cf7f7R73

Reply:
Can we prefix this then?
```suggestion
    throw new Error('invariant: invalid relative URL')
```